crates/dojo-lang/src/compiler.rs

```diff
@@ -118,11 +120,38 @@ impl Compiler for DojoCompiler {
 
         for (decl, class) in zip(contracts, classes) {
             let contract_full_path = decl.module_id().full_path(db.upcast_mut());
+
+            // save expanded contract source file
+            if let Ok(file_id) = db.module_main_file(decl.module_id()) {
+                if let Some(file_content) = db.file_content(file_id) {
+                    let src_file_name = format!("{contract_full_path}.cairo");
+
+                    let mut file =
+                        target_dir.open_rw(src_file_name.clone(), "source file", ws.config())?;
+                    file.write(format_string(db, file_content.to_string()).as_bytes())
+                        .with_context(|| {
+                            format!("failed to serialize contract source: {contract_full_path}")
+                        })?;
+                } else {
+                    return Err(anyhow!("failed to get source file content: {contract_full_path}"));
+                }
+            } else {
+                return Err(anyhow!("failed to get source file: {contract_full_path}"));
+            }
+
+            // save JSON artifact file
             let file_name = format!("{contract_full_path}.json");
+            let mut file = target_dir.open_rw(file_name.clone(), "class file", ws.config())?;
+            serde_json::to_writer_pretty(file.deref_mut(), &class).with_context(|| {
+                format!("failed to serialize contract artifact: {contract_full_path}")
+            })?;
 
-            let mut file = target_dir.open_rw(file_name.clone(), "output file", ws.config())?;
-            serde_json::to_writer_pretty(file.deref_mut(), &class)
-                .with_context(|| format!("failed to serialize contract: {contract_full_path}"))?;
+            // save JSON ABI file
```

Comment:
The compiler already outputs the manifests, where the ABI are already extracted. We should take them from manifests to avoid double generation.

When you will rebase, the manifests folder is still `manifests`. Then we have the profile name, which is available via the workspace with `ws.current_profile()` or via the config `config.profile()`.